libs/sdk-ui-theme-provider/src/cssProperties.ts

```diff
@@ -0,0 +1,132 @@
+// (C) 2020 GoodData Corporation
+import { isObject } from "lodash";
+import { transparentize, darken, lighten } from "polished";
+import { IThemePalette, ITheme } from "@gooddata/sdk-model";
+
+/**
+ *
+ * @param {string} src - Font src
+ * @param {string} number - Font weight
+ */
+function createfontFace(src: string, weight: number): undefined {
+    const newStyle = document.createElement("style").appendChild(
+        document.createTextNode(`
+            @font-face {
+                font-family: GDCustomFont;
+                src: ${src};
+                font-weight: ${weight};
+            }
+        `),
+    );
+    document.head.appendChild(newStyle);
+
+    return undefined; // undefined values are skipped while generating CSS properties
+}
+
+interface ICssProperty {
+    key: string;
+    value: string;
+}
+
+/**
+ * @internal
+ */
+export interface IParserFunction {
+    key: string;
+    fn: (value: any) => string;
+}
```

Comment:
you can make these into types and drop the `I` if you prefer. we try to stick with a rule where:

- interfaces are used for things that MAY be implemented in different ways (by different backend, or different part of code)
- types are used for `struct` like things. where you need to group multiple things and give it a name

not a hard req